### DISCUSSION TITLE
Identity data type

### DISCUSSION THREAD

**Discussion by robbietree8** [ory/kratos#956] [Q&A] [answered]
We are currently build identity server based on kratos since some feature is missing. 
On legacy user system migration, it is good to keep identity id the same as before to minimize migration effort, so we tried to modify the source code of kratos in order to accept identity id from outside in admin api.

In https://github.com/ory/kratos/blob/master/identity/identity.go#L30, it says

> // ID is a unique identifier chosen by you. It can be a URN (e.g. "arn:aws:iam::123456789012"),
// a stringified integer (e.g. "123456789012"), a uuid (e.g. "9f425a8d-7efc-4768-8f23-7647a74fdf13"). It is up to you
// to pick a format you'd like. It is discouraged to use a personally identifiable value here, like the username
// or the email, as this field is immutable.
//
// required: true
ID uuid.UUID `json:"id" faker:"-" db:"id"`

But the data type of ID is `uuid.UUID`, thus it is impossible to set ID to something like stringfield integer, do I understand it in wrong way or is it designed intentionly?

**Comment by aeneasr** [comment, marked as answer]
The documentation is actually not up to date here. We do not support custom IDs at the moment, as this would mess with our plans to add planet-scaling key/value stores (e.g. DynamoDB, Cockroach, ...) in the future.

We are planning on "system" traits which the user him/herself can not modify. The plan would be to have your legacy ID in those fields. Would that work for you?

**Reply by aeneasr** [reply]
I have updated the documentation of the field.

**Reply by robbietree8** [reply]
LGTM
